src/metabase/driver/postgres.clj

```diff
@@ -279,6 +276,20 @@
   (and (str/starts-with? database-type "\"")
        (str/ends-with? database-type "\"")))
 
+(defmethod sql.qp/->honeysql [:postgres :convert-timezone]
+  [driver [_ arg to-tz from-tz]]
+  (let [clause       (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? clause "timestamptz")]
+    (when (and timestamptz? from-tz)
+      (throw (ex-info "`timestamp with time zone` columns shouldn't have a `from timezone`" {:to-tz   to-tz
+                                                                                             :from-tz from-tz})))
+    (let [from-tz (or from-tz (qp.timezone/results-timezone-id))]
```

Comment:
for more detail about why we calling `timezone` twice here, please check my the scenarios I have in the description. 
The steps in each scenario correspond to each time we apply `timezone` here.